HousingManagementSystemApi/Gateways/RepairsHubAlertsGateway.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Amazon.Lambda.Core;
+using HousingManagementSystemApi.Models.RepairsHub;
+
+namespace HousingManagementSystemApi.Gateways
+{
+    public class RepairsHubAlertsGateway : IRepairsHubAlertsGateway
+    {
+        private readonly HttpClient _client;
+
+        public RepairsHubAlertsGateway(IHttpClientFactory httpClientFactory)
+        {
+            _client = httpClientFactory.CreateClient(HttpClientNames.RepairsHubAlerts);
+        }
+
+        public async Task<AlertsViewModel> GetPersonAlerts(string tenancyAgreementReference)
+        {
+            if (string.IsNullOrWhiteSpace(tenancyAgreementReference))
+                throw new ArgumentException(tenancyAgreementReference, nameof(tenancyAgreementReference));
+
+            LambdaLogger.Log("Starting person alert retrieval for tenancy " + tenancyAgreementReference);
+
+            string uri = $"properties/{tenancyAgreementReference}/person-alerts";
+
+            var request = new HttpRequestMessage(HttpMethod.Get, uri);
+
+            LambdaLogger.Log("About to get person alerts at " + uri);
+
+            var response = await _client.SendAsync(request);
+
+            AlertsViewModel alertResult = null;
+
+            if (response.StatusCode == HttpStatusCode.OK)
+            {
+                alertResult = await response.Content.ReadFromJsonAsync<AlertsViewModel>();
+
+                LambdaLogger.Log($"Successfully got {alertResult.Alerts.Count} person alerts for tenancy " + tenancyAgreementReference);
+            }
+            else
+            {
+                LambdaLogger.Log($"Call to {uri} was unsuccessful with error {response.StatusCode}, reason: {response.ReasonPhrase}");
+            }
+
+            return alertResult;

```

Comment:
The same response status logic is used below. Could it be extracted into a method?

Reply:
It's absolutely correct - however - speaking with @adamtry just now, it appears we can remove the person alerts

![image](https://github.com/LBHackney-IT/HousingManagementSystemApi/assets/8542878/5b9a43b6-9a0b-4c61-b8cf-274a9e811684)